### ISSUE TITLE
Eclipse indexer fails with v1.4.x

### ISSUE THREAD

**Issue by gcerretani** [gabime/spdlog#1265]
When switching to v1.4.2 (tested also with v1.4.0 and v1.4.1) Eclipse fails to index the spdlog stuff, even with very simple code, like:
```
#include <spdlog/sinks/stdout_color_sinks.h>

int main() {
	auto console = spdlog::stdout_color_mt("logger");
	console->error("ops");
}
```
I perform **Index -> Rebuild** after each version change. With the example code, the IDE complains about two semantic errors:
-  `Function 'stdout_color_mt' could not be resolved`
-  `Method 'error' could not be resolved`

 Of course the code is compiled without any problem.

Everything is detected correctly with v1.3.1. I'm using _Eclipse CDT version 2019-09 R (4.13.0)_.

**Comment by gcerretani** [comment]
Currently I'm still using the header-only library.

**Comment by tt4g** [comment]
The following sources built successfully with MSVC 2017. I use spdlog 1.4.2 head-only.

```cpp
#include <spdlog/sinks/stdout_color_sinks.h>

int main() {
	auto console = spdlog::stdout_color_mt("logger");

#ifdef SPDLOG_HEADER_ONLY
	console->error("header_only"); // console: [2019-10-11 22:19:11.902] [logger] [error] header_only
#endif

#ifdef SPDLOG_COMPILED_LIB
    console->error("static library"); // no output.
#endif

    return 0;
}

```

I think it's an Eclipse problem because it builds successfully. However, I may not be able to help because I am not familiar with Eclipse.

**Comment by tt4g** [comment]
I found an Eclipse Indexer configuration problem in Stack Overflow: https://stackoverflow.com/questions/44521979/how-to-keep-eclipse-cdt-from-resolving-files-outside-of-the-configured-include-p/44544633#44544633

Does this procedure solve the problem?

**Comment by gcerretani** [comment]
That procedure does not work, but I've tried with some other settings: a tradeoff is to change the indexer configuration is to select `Skip implicit references (e.g. overloaded operators)` in that page. This seems to work also in a much bigger project.

**Comment by gcerretani** [comment]
The problem seems to be fixed with version 1.5.0 (tested with Eclipse CDT 9.10.0.201912051559).

**Comment by gabime** [comment]
Good to know. Thanks for updating
